### PR TITLE
Restart subscription to graph node after an interval

### LINKED ISSUE #173: Observation : Subscription to Graph node stops intermittently after sometime
**Observation**:
After starting Facilitator, Graph node subscription stops after interval X (observation is X=15-20 minutes). 

Discussed different solutions with TheGraph support on discord. Graph node doesn't have KeepAlive support. This helps to keep alive connection.

Solution suggested by TheGraph is restart the subscription after interval X.

### PR DESCRIPTION
**Pull request by abhayks1**
Observation:
After starting Facilitator, Graph node subscription stops after interval X (observation is X=10-15 minutes).

Discussed different solutions with TheGraph support on discord. Graph node doesn't have KeepAlive support. This helps to keep alive connection.

Solution suggested by TheGraph is restart the subscription after interval X.

Fixes #173

### CODE REVIEW

File: src/Facilitator.ts [hunk 1]
```diff
@@ -9,29 +9,49 @@ export default class Facilitator {
...
 
   /** Starts the facilitator by subscribing to subscription queries. */
   public async start(): Promise<void> {
-    Logger.info('Starting subscription to block chain events');
-    await this.originSubscriber.subscribe();
-    Logger.info('Subscription to origin block chain is done');
-    await this.auxiliarySubscriber.subscribe();
-    Logger.info('Subscription to auxiliary block chain is done');
+    this.subscribeToSubGraphs();
+    setInterval(() => this.restartSubscription(), this.subscriptionRestartTime);
```
Comment (gulshanvasnani) [inline src/Facilitator.ts]: Please clear the interval when facilitator stops.

File: src/Facilitator.ts [hunk 2]
```diff
@@ -9,29 +9,49 @@ export default class Facilitator {
...
    * This function should be called on signint or control-c.
    */
   public async stop(): Promise<void> {
+    this.unsubscribeToSubGraphs();
+  }
+
+  // It restarts the subscription
+  private async restartSubscription() {
+    this.unsubscribeToSubGraphs();
+    this.subscribeToSubGraphs()
```
Comment (deepesh-kn) [inline src/Facilitator.ts]: Add await, also in stop function.

File: src/Facilitator.ts [hunk 3]
```diff
@@ -9,29 +9,49 @@ export default class Facilitator {
...
   }
 
   /** Starts the facilitator by subscribing to subscription queries. */
   public async start(): Promise<void> {
-    Logger.info('Starting subscription to block chain events');
-    await this.originSubscriber.subscribe();
-    Logger.info('Subscription to origin block chain is done');
-    await this.auxiliarySubscriber.subscribe();
-    Logger.info('Subscription to auxiliary block chain is done');
+    this.subscribeToSubGraphs();
```
Comment (deepesh-kn) [inline src/Facilitator.ts]:
```suggestion
    await this.subscribeToSubGraphs();
```

File: src/Facilitator.ts [hunk 4]
```diff
@@ -9,29 +9,49 @@ export default class Facilitator {
...
+  private subscriptionRestartTime: number;
+
   /**
    * @param originSubscriber Origin subscriber instance.
    * @param auxiliarySubscriber Auxiliary subscriber instance.
    */
   public constructor(originSubscriber: Subscriber, auxiliarySubscriber: Subscriber) {
     this.originSubscriber = originSubscriber;
     this.auxiliarySubscriber = auxiliarySubscriber;
+    this.subscriptionRestartTime = 10*60*1000; // 10 minutes
```
Comment (deepesh-kn) [inline src/Facilitator.ts]: Use a constant variable for the value

File: src/Facilitator.ts [hunk 5]
```diff
@@ -9,29 +9,49 @@ export default class Facilitator {
 
   private auxiliarySubscriber: Subscriber;
 
+  private subscriptionRestartTime: number;
```
Comment (deepesh-kn) [inline src/Facilitator.ts]:
```suggestion
  private subscriptionRestartDuration: number;
```

File: src/Facilitator.ts (hunk 1 above)
Comment (abhayks1) [reply src/Facilitator.ts]: Done!

File: src/Facilitator.ts (hunk 2 above)
Comment (abhayks1) [reply src/Facilitator.ts]: Done!

File: src/Facilitator.ts (hunk 3 above)
Comment (abhayks1) [reply src/Facilitator.ts]: Added.

File: src/Facilitator.ts (hunk 5 above)
Comment (abhayks1) [reply src/Facilitator.ts]: Done!

File: src/Facilitator.ts (hunk 4 above)
Comment (abhayks1) [reply src/Facilitator.ts]: Done!

**Comment by abhayks1** [comment]
Available for review 🐰

File: src/Facilitator.ts [hunk 6]
```diff
@@ -9,29 +10,53 @@ export default class Facilitator {
...
 
   /** Starts the facilitator by subscribing to subscription queries. */
   public async start(): Promise<void> {
-    Logger.info('Starting subscription to block chain events');
-    await this.originSubscriber.subscribe();
-    Logger.info('Subscription to origin block chain is done');
-    await this.auxiliarySubscriber.subscribe();
-    Logger.info('Subscription to auxiliary block chain is done');
+    this.subscribeToSubGraphs();
+    this.handle = setInterval(
```
Comment (gulshanvasnani) [inline src/Facilitator.ts]: I would suggest name for the interval as **restartSubscriptionInterval**

**Pull request merged**